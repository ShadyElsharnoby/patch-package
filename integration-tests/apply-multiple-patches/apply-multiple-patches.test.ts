import { runIntegrationTest } from "../runIntegrationTest"
runIntegrationTest({
  projectName: "apply-multiple-patches",
  shouldProduceSnapshots: true,
})
