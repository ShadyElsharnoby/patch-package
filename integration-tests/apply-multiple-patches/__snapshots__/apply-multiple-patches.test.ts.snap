// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test apply-multiple-patches: 00: patch-package happily applies all three good patches 1`] = `
"SNAPSHOT: patch-package happily applies all three good patches
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔
left-pad@1.3.0 (3 world) ✔
left-pad@1.3.0 (4 goodbye) ✔
END SNAPSHOT"
`;

exports[`Test apply-multiple-patches: 01: patch-package stores a state file to list the patches that have been applied 1`] = `
"SNAPSHOT: patch-package stores a state file to list the patches that have been applied
{
    \\"isRebasing\\": false,
    \\"patches\\": [
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"404c604ed830db6a0605f86cb9165ced136848f70986b23bf877bcf38968c1c9\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+001+hello.patch\\"
        },
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"f2859c7193de8d9578bdde7e226de516adc8d972d6e76997cbe1f41b1a535359\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+003+world.patch\\"
        },
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"946d4e578decc1e475ca9b0de07353791969312fd2bf5d9cfc5182b86d9804ad\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+004+goodbye.patch\\"
        }
    ],
    \\"version\\": 1
}END SNAPSHOT"
`;

exports[`Test apply-multiple-patches: 02: patch-package fails when a patch in the sequence is invalid 1`] = `
"SNAPSHOT: patch-package fails when a patch in the sequence is invalid
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔

⛔ ERROR

Failed to apply patch file left-pad+1.3.0+002+broken.patch.

If this patch file is no longer useful, delete it and run

  patch-package

To partially apply the patch (if possible) and output a log of errors to fix, run

  patch-package --partial

After which you should make any required changes inside node_modules/left-pad, and finally run

  patch-package left-pad

to update the patch file.

END SNAPSHOT"
`;

exports[`Test apply-multiple-patches: 03: patch-package only applies the first patch if the second of three is invalid 1`] = `
"SNAPSHOT: patch-package only applies the first patch if the second of three is invalid
patch-package 0.0.0
Applying patches...
left-pad@1.3.0 (1 hello) ✔

⛔ ERROR

Failed to apply patch file left-pad+1.3.0+002+broken.patch.

If this patch file is no longer useful, delete it and run

  patch-package

To partially apply the patch (if possible) and output a log of errors to fix, run

  patch-package --partial

After which you should make any required changes inside node_modules/left-pad, and finally run

  patch-package left-pad

to update the patch file.

END SNAPSHOT"
`;

exports[`Test apply-multiple-patches: 04: patch-package stores a state file of only the first patch if there was an error 1`] = `
"SNAPSHOT: patch-package stores a state file of only the first patch if there was an error
{
    \\"isRebasing\\": true,
    \\"patches\\": [
        {
            \\"didApply\\": true,
            \\"patchContentHash\\": \\"404c604ed830db6a0605f86cb9165ced136848f70986b23bf877bcf38968c1c9\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+001+hello.patch\\"
        },
        {
            \\"didApply\\": false,
            \\"patchContentHash\\": \\"9c5c141e2578b4178fd57dd7726488c2f7eab32e23a7848701da29dcb371b9f2\\",
            \\"patchFilename\\": \\"left-pad+1.3.0+002+broken.patch\\"
        }
    ],
    \\"version\\": 1
}END SNAPSHOT"
`;

exports[`Test apply-multiple-patches: 05: patch-package fails when a patch file is removed 1`] = `
"SNAPSHOT: patch-package fails when a patch file is removed
patch-package 0.0.0
Applying patches...
Error: The patches for left-pad have changed. You should reinstall your node_modules folder to make sure the package is up to date
END SNAPSHOT"
`;
